mitodl/mit-open
# fix flaky tests

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5000

### Description (What does it do?)
This PR fixes some issues with JS tests that have been popping up:

 - The `EditChannelPage` component renders within `ChannelPage`, which if the channel type is `unit`, renders testimonials. The tests calling `EditChannelPage` did not mock the testimonial API endpoints, therefore causing errors. These endpoints are now mocked.
 - Tests looking at variants of `LearningResourceCard` were failing to find images by role `"img"`, so I put a `data-testid` attribute on the `Card`'s `Image` element and looked it up by that, which worked. I'm not sure if this is an issue with react testing library, as the image element is shown as being found when the test fails as an accessible element and all `img` elements should automatically have the role of `img`.

### How can this be tested?
 - Check out this branch and spin up `mit-open`
 - Run `docker compose exec watch yarn test` several times, making sure it passes each time
